tests/dem/full_contact_functions.h

```diff
@@ -0,0 +1,339 @@
+  Tensor<1, 3>       velocity_0;
+  Tensor<1, 3>       velocity_1;
+  Tensor<1, 3>       omega_0;
+  Tensor<1, 3>       omega_1;
+  Tensor<1, 3>       normal_unit_vector;
+  Tensor<1, 3>       relative_velocity;
+  Tensor<1, 3>       tangential_relative_velocity;
+  const double       force_calculation_threshold_distance =
+    cut_off_factor * 0.5 * (p.diameter[0] + p.diameter[1]);
+
+  // Open file and write names of columns
+  std::ofstream file(filename + ".dat", std::ios::binary);
+  file
+    << "time force_0_x force_0_y force_0_z torque_0_x torque_0_y torque_0_z normal_overlap tangential_overlap_x tangential_overlap_y tangential_overlap_z velocity_0_x velocity_0_y velocity_0_z velocity_1_x velocity_1_y velocity_1_z omega_0_x omega_0_y omega_0_z omega_1_x omega_1_y omega_1_z"
+    << "\n";
+
+  for (unsigned int iteration = 0; iteration < max_iteration; ++iteration)
+    {
+      // Reinitializing forces
+      reinitialize_force(particle_handler, torque, force);
+
+      contact_manager.update_local_particles_in_cells(particle_handler);
+
+      // Dummy Adaptive sparse contacts object and particle-particle broad
+      // search
+      AdaptiveSparseContacts<dim, PropertiesIndex>
+        dummy_adaptive_sparse_contacts;
+      contact_manager.execute_particle_particle_broad_search(
+        particle_handler, dummy_adaptive_sparse_contacts);
+
+      // Calling fine search
+      contact_manager.execute_particle_particle_fine_search(
+        neighborhood_threshold);
+
+      // Calling force calculation
+      force_object.calculate_particle_particle_contact_force(
+        contact_manager.get_local_adjacent_particles(),
+        contact_manager.get_ghost_adjacent_particles(),
+        contact_manager.get_local_local_periodic_adjacent_particles(),
+        contact_manager.get_local_ghost_periodic_adjacent_particles(),
+        contact_manager.get_ghost_local_periodic_adjacent_particles(),
+        dt,
+        torque,
+        force);
+
+      distance = (position_0).distance(position_1);
+
+      // Checking if contact is still ongoing (overlap superior to force
+      // calculation threshold)
+      if (normal_overlap > force_calculation_threshold_distance &&
+          (0.5 * (p.diameter[0] + p.diameter[1]) - distance) <=
+            force_calculation_threshold_distance)
+        {
+          contact_is_ongoing = false;
+        }
+
+      // Calculating overlap and tangential overlap
+      update_velocities<PropertiesIndex>(particle_0->get_properties(),
+                                         particle_1->get_properties(),
+                                         velocity_0,
+                                         velocity_1,
+                                         omega_0,
+                                         omega_1);
+      normal_overlap = 0.5 * (p.diameter[0] + p.diameter[1]) - distance;
+      normal_unit_vector =
+        (position_1 - position_0) / (position_1 - position_0).norm();
```

Comment:
I think you can do this: 

```suggestion
        (position_1 - position_0) / distance;
```